internal/strings/strings.go

```diff
@@ -106,3 +106,28 @@ func InSlice(a string, list []string) bool {
 func WrapUnsafe(buf []byte) string {
 	return *(*string)(unsafe.Pointer(&buf))
 }
+
+// AsciiToLower converts ASCII characters in the string to lowercase
+// without allocating additional memory, unlike strings.ToLower.
+func AsciiToLower(s string) string {
+	for i := 0; i < len(s); i++ {
+		if s[i] >= 'A' && s[i] <= 'Z' {
+			return asciiToLowerInPlace(s, i)
+		}
+	}
+	return s
+}
+
+// asciiToLowerInPlace converts ASCII characters in the string to lowercase
+// starting from the specified index.
+func asciiToLowerInPlace(s string, start int) string {
+	res := []byte(s)
```

Comment:
Not sure if you have access to medium but this can be done more efficiently https://josestg.medium.com/140x-faster-string-to-byte-and-byte-to-string-conversions-with-zero-allocation-in-go-200b4d7105fc